datumaro/plugins/voc_format/extractor.py

```diff
@@ -344,31 +405,36 @@ def _load_annotations(self, item_id):
         if osp.isfile(inst_path):
             instances_mask = lazy_mask(inst_path, _inverse_inst_colormap)
 
+        label_cat = self._categories[AnnotationType.label]
+
         if instances_mask is not None:
             compiled_mask = CompiledMask(class_mask, instances_mask)
 
-            label_cat = self._categories[AnnotationType.label]
-
             if class_mask is not None:
                 instance_labels = compiled_mask.get_instance_labels()
             else:
                 instance_labels = {i: None for i in range(compiled_mask.instance_count)}
 
             for instance_id, label_id in instance_labels.items():
                 if len(label_cat) <= label_id:
-                    raise Exception(
-                        "Item %s: a mask has unexpected class number %s" % (item_id, label_id)
+                    self._ctx.error_policy.report_annotation_error(
+                        UndeclaredLabelError(str(label_id)), item_id=(item_id, self._subset)
                     )
 
                 image = compiled_mask.lazy_extract(instance_id)
 
                 item_annotations.append(Mask(image=image, label=label_id, group=instance_id))
         elif class_mask is not None:
-            log.warning("Item %s: only class segmentations available" % item_id)
+            log.warning(f"Item {item_id}: only class segmentations available")
```

Comment:
I think with logging it's better to do this:

```suggestion
            log.warning("Item %s: only class segmentations available", item_id)
```

since that prevents the message from being formatted unless the appropriate log level is enabled.

Reply:
Applied